dex/testing/firo/README_ELECTRUM_HARNESSES.md

```diff
@@ -0,0 +1,130 @@
+and are unsupported by their respective maintainers on PyPI. Git should be latest.Pip will be re-downloaded and it's version upgraded by the script into a virtual environment each time.
+
+### Architecture
+
+- ELECTRUM WALLET CLIENT
+- ELECTRUMX SERVER
+- FIRO CHAIN HARNESS
+
+All three scripts store data in **~/dextest/...** dirsectory tree.
+
+## Using
+
+You must have `firod` and `firo-cli` in `PATH` to use the chain server harness. Use 3 tty's
+and run each line below in a **separate** tty:
+
+```bash
+$  ./harness.sh
+$  ./electrumx.sh
+$  ./electrum.sh
+```
+
+The Electrum-Firo wallet client will have a prepared, empty but encrypted
+regtest wallet.
+
+Password is "abc".
+
+## Development
+
+### Firo Chain Server
+
+For the Firo chain server see the README_HARNESS.md in this directory.
+
+### Server
+
+The **electrumx.sh** script first cleans part of the **~/dextest/electrum/firo/server...**
+directory tree.
+
+The script then downloads a specific commit from:
+<https://github.com/firoorg/electrumx-firo.git> to the .../server/electrumx-repo directory.
+
+It then creates a python virtual environment (venv) sandbox and installs the
+requesting python3 interpreter, latest version of pip and all required modules
+in `setup.py` into the sandbox.
+If any need building from C source it will be done at this point.
+
+Server certificates are created. Then all the required environment variables to
+configure the server exported; and the electrumX daemon started
+
+The electrumX daemon will connect to the firo chain harness node **alpha**
+
+### Client
+
+The **electrum.sh** script first cleans part of the **~/dextest/electrum/firo/client...**
+directory tree.
+
+The script then downloads a specific commit from:
+ <https://github.com/firoorg/electrum-firo.git> to the ../client/electrum-repo directory.
+
+It then creates a python virtual environment (venv) sandbox and installs the
+requesting python3 interpreter, latest version of pip and all required modules
+from files in `contrib/requirements` directory into the sandbox. If any need building from C source it will be done at
+this point.
+
+A prepared, empty but encrypted electrum wallet is copied to the electrum data
+directory at **~/dextest/electrum/firo/client/wallet/regtest/wallets** directory.
+
+By default the electrum client is started as a daemon and the default wallet is loaded.
+Use **stop-daemon** script to stop the daemon.
+
+The electrum client wallet will connect to the firo electrumX daemon.
+
+### Simnet Trade Testing
+
+In seperate tty's start up:
+
+   - btc harness
+   - dcr harness
+   - firo harness
+   - electrumx.sh Electrum-Firo regtest server
+   - electrum.sh Electrum-Firo regtest wallet
+   - dcrdex harness
+
+Ensure dcrctl, firod and firo-cli are in PATH for simnet-trade-tests exec process
+mining.
+
+```bash
+$  ./run dcrfiroelectrum --runonce --all
+```
+
+Best results by nuking everything after each run.
```

Comment:
There are 7 scripts running here and several test cases. It is possible for the _firo chain harness_ and particularly the _firo_electrumX_ test server to get out of sync in the blocks they think the have. Best to tear down after each set of full simnet tests.

I guess this is more of a human forgetfulness issue rather than which is why I made a start on automating `simnet-trade-test` setup and teardown [here](https://github.com/dev-warrior777/fastup)

It does not extend to electrum scripts yet though